src/eradiate/scenes/illumination/_spot.py

```diff
@@ -0,0 +1,217 @@
+            units=ucc.deferred("length"),
+        ),
+        doc="Point location targeted by the spot.\n"
+        "\n"
+        "Unit-enabled field (default: ucc['length']).",
+        type="quantity",
+        init_type="array-like",
+        default="[0, 0, 0] m",
+    )
+
+    @target.validator
+    @origin.validator
+    def _target_origin_validator(self, attribute, value):
+        if np.allclose(self.target, self.origin):
+            raise ValueError(
+                f"While initializing {attribute}: "
+                f"Origin and target must not be equal, "
+                f"got target = {self.target}, origin = {self.origin}"
+            )
+
+    up: np.ndarray = documented(
+        attrs.field(
+            factory=lambda: [0, 0, 1],
+            converter=np.array,
+            validator=validators.has_len(3),
+        ),
+        doc="A 3-vector specifying the up direction of the spot.\n"
+        "This vector must be different from the spots's pointing direction,\n"
+        "which is given by ``target - origin``.",
+        type="array",
+        default="[0, 0, 1]",
+    )
+
+    @up.validator
+    def _up_validator(self, attribute, value):
+        direction = self.target - self.origin
+        if np.allclose(np.cross(direction, value), 0):
+            raise ValueError(
+                f"While initializing '{attribute.name}': "
+                f"up direction must not be colinear with viewing direction, "
+                f"got up = {self.up}, direction = {direction}"
+            )
+
+    beam_width: pint.Quantity = documented(
+        pinttr.field(default=10.0 * ureg.deg, units=ureg.deg),
+        doc="Spot light beam width.\n\nUnit-enabled field (default: degree).",
+        type="quantity",
+        init_type="quantity or float",
+        default="10°",
+    )
+
+    intensity: Spectrum = documented(
+        attrs.field(
+            default=1.0,
+            converter=spectrum_factory.converter("intensity"),
+            validator=[
+                attrs.validators.instance_of(Spectrum),
+                has_quantity("intensity"),
+            ],
+        ),
+        doc="Emitted power in the plane orthogonal to the illumination direction. "
+        "Must be an intensity spectrum (in W/sr/nm or compatible unit). "
+        "Can be initialised with a dictionary processed by "
+        ":meth:`.SpectrumFactory.convert`.",
+        type=":class:`~eradiate.scenes.spectra.Spectrum`",
+        init_type=":class:`~eradiate.scenes.spectra.Spectrum` or dict or float",
+        default="1.0 ucc[intensity]",
+    )
+
+    beam_profile: t.Optional[Path] = documented(
+        attrs.field(
+            default=None,
+            converter=attrs.converters.optional(Path)
+        ),
+        type="Path",
+        init_type="path-like",
+        default=None,
+        doc="Path to the beam profile file, relative to the current working directory. "
+            "Must be a valid bitmap image file.",
```

Comment:
I don't like that idea. We'd either have to make arbitrary choices or have a dozen profiles in the data store.
- Gaussian 1-sigma
- Gaussian 2-sigma
- Gaussian 3-sigma
- Gaussian 0.5-sigma
...

And the first user to use it will of course want something we don't offer... I think the default should just be untextured.

Reply:
* It's always nice to ship a bit of data to demonstrate features; with that in mind, the data store becomes the natural entry point.
* I'd be in favour of shipping a single default profile as an example.
* Having built-in profiles doesn't mean constraining users. We ship SRFs, but users who want to provide their own can as well and they have examples and documentation to prepare their data.
* I agree with leaving the default untextured.